### PR TITLE
Button: deprecated textColor prop values

### PR DESCRIPTION
**Pull request by AlbertCarreras**
- Deprecated `textColor` prop color values used for `white` & `transparent` background buttons. White/Transparent backgrounds default to `darkGray`.
- New color value `color = 'transparentWhiteText'` for transparent background buttons with white text 
image

## Before
image
image

## After
image
image

# Codemod

## From
```
       <Box>
      <Button text="Medium-sized button" inline textColor="red"/>
      <Button text="Medium-sized button" inline color="transparent"/>
      <Button text="Medium-sized button" inline textColor="white" color="transparent" />
      <Button text="Medium-sized button" inline textColor="white" color="red"/>
      <Renamed text="Medium-sized button" inline textColor="red"/>
    </Box>
```
## To
```
    <Box>
      <Button text="Medium-sized button" inline />
      <Button text="Medium-sized button" inline color="transparent"/>
      <Button text="Medium-sized button" inline color="transparentWhiteText" />
      <Button text="Medium-sized button" inline color="red" />
      <Renamed text="Medium-sized button" inline />
    </Box>
```

Codemod helper, run
yarn codemod --parser=flow -t=packages/gestalt-codemods/12.6.0-12.7.0/button-remove-textColor-prop.js relative/path/to/your/code

### CODE REVIEW

File: packages/gestalt/src/Button.js [hunk 1]
```diff
@@ -247,7 +235,7 @@ const ButtonWithForwardRef: React$AbstractComponent<
         onTouchMove={handleTouchMove}
         onTouchStart={handleTouchStart}
         ref={innerRef}
-        style={compressStyle}
+        style={compressStyle || undefined}
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:238]: @AlbertCarreras why this change?

File: packages/gestalt/src/Button.js [hunk 2]
```diff
@@ -44,7 +44,7 @@ type BaseButton = {|
   >,
   size?: 'sm' | 'md' | 'lg',
   text: string,
-  textColor?: 'white' | 'darkGray' | 'blue' | 'red',
+  textColor?: 'white',
```
Comment (christianvuerings) [inline packages/gestalt/src/Button.js:47]: @AlbertCarreras why do we have to keep this?

Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:47]: image
image

Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:47]: I'm gonna document

File: packages/gestalt/src/Button.js (hunk 1 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Button.js:238]: Because it's null, therefore some snapshots show `style='null'`  
image

**Review by christianvuerings — APPROVED** [review]
